Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift

```diff
@@ -145,38 +199,64 @@ import WebKitSwift
         }
     }
 
-    @objc(restoreSelectionAcceptedReplacements:completion:)
-    public func restoreSelection(acceptedReplacements: Bool) async {
+    func restoreSelectionAcceptedReplacements(_ acceptedReplacements: Bool, completion: @MainActor @Sendable @escaping () -> Void) {
+        Task {
+            await restoreSelectionAsync(acceptedReplacements: acceptedReplacements)
+            completion()
+        }
+    }
+
+    @nonobjc
+    private final func restoreSelectionAsync(acceptedReplacements: Bool) async {
         guard let contextRange = self.contextRange else {
             assertionFailure()
             return
         }
 
-        let range = acceptedReplacements ? contextRange.lowerBound..<(contextRange.upperBound + self.processedRangeOffset) : contextRange;
+        let range = acceptedReplacements ? contextRange.lowerBound..<(contextRange.upperBound + self.processedRangeOffset) : contextRange
         await self.delegate.intelligenceTextEffectCoordinator(self, setSelectionFor: NSRange(range))
     }
 
-    @objc(hideEffectsWithCompletion:)
-    public func hideEffects() async {
+    func hideEffects(completion: @MainActor @Sendable @escaping () -> Void) {
+        Task {
+            await hideEffectsAsync()
+            completion()
+        }
+    }
+
+    @nonobjc
+    private final func hideEffectsAsync() async {
         await self.viewManager.hideEffects()
     }
 
-    @objc(showEffectsWithCompletion:)
-    public func showEffects() async {
+    func showEffects(completion: @MainActor @Sendable @escaping () -> Void) {
+        Task {
```

Comment:
Please explain the introduction of `Task` here and elsewhere in the commit message.